### ISSUE TITLE
Incorrect "Meets Eligibility" tag for applicants who were never shown Eligibility question 

### ISSUE THREAD

**Issue by swatkat1** [civiform/civiform#5614]
**Describe the bug**
If an applicant starts on an old version of the program which doesn't include eligibility question (the newer version has Eligibility question), later submits it, the Program Admin sees "Meets Eligibility" tag even though they haven't answered any Eligibility question. 

**To Reproduce**

1. Create a program with a single question and publish it.
2. Login as an applicant and start the application without submitting it.
3. Login as a CiviForm admin and add an Eligibility question and publish again.
4. Login as the applicant and submit the application. The applicant is not shown the Eligibility question.
5. Login as the Program Admin and check "Applications". We see the "Meets Eligibility" tag for the applicant.

**Expected behavior**
"Meets Eligibility" should be shown only when the candidate answers the Eligibility question and passes it. Else incorrect tagging will create confusion.

**Screenshots**
 Example: After adding the condition: 
![image](https://github.com/civiform/civiform/assets/101214724/8b779f83-7b66-4bd9-81f8-36b9b10c703b) and submitting it: 
![image](https://github.com/civiform/civiform/assets/101214724/1d717a6b-e783-4947-855c-9b3dbebf421f)
The Program admin sees this: 
![image](https://github.com/civiform/civiform/assets/101214724/81f1ec50-2d8a-4641-ac7e-78419039b0d1)

Relates to #5541

**Comment by elisekalstad** [comment]
Reviewed at 9/14 cross team meeting: consider short term solution to NOT show "meets eligibility" IF there is no data under eligiblity questions. 

Long term solution likely tied to updating version of program when application is in draft state. ("Update to the latest program application! #5541 )

**Comment by dkatzz** [comment]
FYI the part of the code is here https://github.com/civiform/civiform/blob/1509a736a23952a05b8d808d8a0bb4be20ae4a7a/server/app/views/admin/programs/ProgramApplicationListView.java#L118. It is looking at whether eligibility is enabled for the most recent program version, not the version of the program when the application was submitted.

**Comment by tallulahkay** [comment]
@gwendolyngoetz is this still an issue? it looks like you may have fixed it because this issue is closed: https://github.com/civiform/civiform/issues/5541

**Comment by gwendolyngoetz** [comment]
@tallulahkay yes, fast forwarding feature would have resolved this. Going to close.
